pycbc/tmpltbank/lambda_mapping.py

```diff
@@ -193,43 +201,90 @@ def get_chirp_params_new(mass1, mass2, spin1z, spin2z, f0, order):
         mass2 = numpy.array([mass2])
         spin1z = numpy.array([spin1z])
         spin2z = numpy.array([spin2z])
+        if quadparam1 is not None:
+            quadparam1 = numpy.array([quadparam1])
+        if quadparam2 is not None:
+            quadparam2 = numpy.array([quadparam2])
+        if lambda1 is not None:
+            lambda1 = numpy.array([lambda1])
+        if lambda2 is not None:
+            lambda2 = numpy.array([lambda2])
         num_points = 1
-    lal_pars = CreateDict()
-    phasing_vs = numpy.zeros([num_points, 13])
-    phasing_vlogvs = numpy.zeros([num_points, 13])
-    phasing_vlogvsqs = numpy.zeros([num_points, 13])
-    for i in xrange(num_points):
-        phasing = lalsimulation.SimInspiralTaylorF2AlignedPhasing(
-                            mass1[i], mass2[i], spin1z[i], spin2z[i], lal_pars)
-        phasing_vs[i] = phasing.v
-        phasing_vlogvs[i] = phasing.vlogv
-        phasing_vlogvsqs[i] = phasing.vlogvsq
 
-    pmf = PI * (mass1 + mass2)*MTSUN_SI * f0
+    if quadparam1 is None:
+        quadparam1 = numpy.zeros(len(mass1), dtype=float)
+    if quadparam2 is None:
+        quadparam2 = numpy.zeros(len(mass1), dtype=float)
+    if lambda1 is None:
+        lambda1 = numpy.zeros(len(mass1), dtype=float)
+    if lambda2 is None:
+        lambda2 = numpy.zeros(len(mass1), dtype=float)
+
+    mass1_v = CreateREAL8Vector(len(mass1))
```

Comment:
Would this code look better by turning `mass*` into PyCBC Arrays and using their `lal()` attributes in the call to `SimInspiralTaylorF2AlignedPhasingArray()`? Or do they have to be precisely `REAL8Vector`s?

Reply:
I don't think that would work. The LAL methods convert to Time or Frequency series, which these are not. The lal function requires REAL8Vector s as input.

Reply:
@titodalcanton Is correct. There is a lal method for the standard pycbc Array that converts to REAL8Vector if the dtype is float64 for example.

Reply:
ex. 
a = pycbc.types.Array([0, 1, 2, 3], dtype=numpy.float64)

b = a.lal()

Reply:
Alright, I'll try this out.

Reply:
Is there a speed penalty in converting from a numpy array to a PyCBC array with the same dtype?

Reply:
Trying to change this is becoming rather painful. If I make the arrays PyCBC arrays I repeatedly hit:
```
TypeError: unsupported operand type(s) for *: 'ArrayWithAligned' and 'Array'
```
Why is multiplication such a hard operation for a slice of a PyCBC Array!?

This is making me want to not use PyCBC arrays at all here, and just write an inline function to convert numpy array -> LAL array (or just ignoring this).

Reply:
@spxiwh I can try to look at this if you have an example. It looks like you tried to multiple an Array by and numpy array ("ArraywithAligned" is an numpy.ndarray).

Reply:
Here's a minimal example to reproduce the base problem (which gives a different error, but I had started using Array.data to avoid this). I guess the problem is that I want to use 2D arrays:
```
b = pycbc.types.Array([1,2,3,4,5,6,7,8,9,9])
c = pycbc.types.Array([[1,2,3,4,5,6,7,8,9,9],[1,2,3,4,5,6,7,8,9,9]])
c[0,1:3] = b[1:3]
```